crates/extensions/c8y_mapper_ext/src/alarm_converter.rs

```diff
@@ -110,6 +167,15 @@ impl AlarmConverter {
         }
     }
 
+    pub(crate) fn new_get_c8y_alarm_topic(&self, topic: &str) -> Result<String, ConversionError> {
+        let topic_split: Vec<&str> = topic.split('/').collect();
+        if topic_split[2].eq("main") {
```

Comment:
Why 2? I thought the `topic` had something like `te/device/main///a/alarm_type`.

Reply:
Yes, here the index of `main` is  `2`.